### PR TITLE
Fast stat bucket changes

### PR DESCRIPTION
**Pull request by ankitaluthra1**
### Description

- Adds new struct folder
- Updates stat cache to use new struct folder
- Updates Get folder to use new struct folder

currently these methods are not being used in any flow

### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - NA
2. Unit tests - Done
3. Integration tests - Done

### CODE REVIEW

**Review by Tulsishah — APPROVED** [review]

File: internal/cache/metadata/stat_cache.go [hunk 1]
```diff
@@ -18,7 +18,6 @@ import (
 	"math"
 	"time"
 
-	"cloud.google.com/go/storage/control/apiv2/controlpb"
```
Comment (charith87) [inline internal/cache/metadata/stat_cache.go:21]: Is the old one not used anymore?

Comment (charith87) [reply internal/cache/metadata/stat_cache.go:21]: What about non HNS buckets? Would this not affect there.

Comment (Tulsishah) [reply internal/cache/metadata/stat_cache.go:21]: This is for folders only. We have created separate folder structure [here](https://github.com/GoogleCloudPlatform/gcsfuse/tree/master/internal/storage/gcs) for controlpb.folder to enable the addition of custom data in the future and to decouple folder struct in gcsfuse code from control package struct.

Comment (ankitaluthra1) [reply internal/cache/metadata/stat_cache.go:21]: yes this is just a small refactoring to have a our own struct for decoupling our methods from sdk folder

**Pull request merged**